doc/rtd/topics/code_review.rst

```diff
@@ -0,0 +1,170 @@
+   This could be a core developer or community member.  (This is likely
+   the role you will be taking in this process.)
+
+the Committers
+   The group of cloud-init core developers.
+
+   In this process doc, cloud-init core developers are treated as a
+   bloc of interchangeable people.  This is a simplification, as in
+   reality we will need to manage transfer of reviews, dismissal of
+   stale reviews, parallel reviews from multiple committers, etc.  If
+   we find that handling these on an ad-hoc basis is causing problems,
+   then we should revisit and expand this process to include them.
+
+   (If the **Proposer** is a core developer, then they are not
+   considered part of this group for the purposes of this process.)
+
+Opening Phase
+=============
+
+In this phase, the **Proposer** is responsible for all actions. They
+should get a PR into good enough shape that it is worth **the
+Committers** spending time reviewing it.  Specifically, they are
+responsible for getting to a point where the continuous integration
+(CI) testing in Travis is passing.  Once a PR is passing CI, it moves
+into the :ref:`Review phase`.
+
+These are the steps that comprise the opening phase:
+
+1. The **Proposer** opens PR
+
+2. An automated comment will be added to the PR, outlining the steps
+   expected of the **Proposer** before a **Committer** will review
+   their PR.  (N.B. This automation is not yet implemented!)
+
+3. CI runs automatically, and if
+
+   CI fails
+      The **Proposer** is expected to fix CI failures.  If the
+      **Proposer** doesn't understand the nature of the failures they
+      are seeing, they should comment in the pull request to request
+      assistance.  Alternatively, for more immediate assistance, they
+      can ask in ``#cloud-init`` on the Freenode IRC network.
+
+   CI passes
+      Move on to the :ref:`Review phase`.
+
+Review Phase
+============
+
+In this phase, the **Proposer** and the **Committers** will iterate
+together to, hopefully, get the PR merged into the cloud-init codebase.
+There are three potential outcomes: merged, rejected permanently, and
+temporarily closed.
+
+These are the steps that comprise the review phase:
+
+1. **The Committers** assign a **Committer** to the PR
+
+   The cloud-init core developers will work amongst themselves to
+   ensure that this happens in a timely fashion.  (Generally,
+   assignment will happen at the team's daily internal meeting.)  Note
+   that, in GitHub terms, this is setting an Assignee, not requesting a
+   review.
+
+2. That **Committer** reviews the PR, resulting in one of the
+   following:
+
+   Approve
+     The **Committer** will squash merge immediately.
+
+     There may be circumstances where a PR should not be merged
+     immediately; these will be handled case-by-case, and the
+     **Proposer** should make this requirement very clear in their pull
+     request description.
+
```

Comment:
Well there may also be the case where a core team member knows of something in flight and the PR should be held until after the in flight work is complete. This case should be documented here as well.

Reply:
Agreed.